sklearn/utils/extmath.py

```diff
@@ -287,11 +287,11 @@ def randomized_svd(M, n_components, n_oversamples=10, n_iter=None,
         number can improve speed but can negatively impact the quality of
         approximation of singular vectors and singular values.
 
-    n_iter: int (default is 4)
+    n_iter: int or 'auto' (default is 'auto')
         Number of power iterations. It can be used to deal with very noisy
-        problems. When `n_components` is small (< .1 * min(X.shape)) `n_iter`
-        is set to 7, unless the user specifies a higher number. This improves
-        precision with few components.
+        problems. When 'auto', it is set to 4, unless `n_components` is small
+        (< .1 * min(X.shape)) `n_iter` in which case is set to 7.
```

Comment:
Maybe the following would be clearer:

```
When `n_iter='auto'`, it is effectively set to 4 to when `n_components` is larger than `0.1 * min(X.shape)` and 7 otherwise.
```